docs/reference/schemas/config/functions/secret.md

```diff
@@ -0,0 +1,177 @@
+---
+description: Reference for the 'secret' DSC configuration document function
+ms.date:     08/22/2025
+ms.topic:    reference
+title:       secret
+---
+
+# secret
+
+## Synopsis
+
+Retrieves secrets from registered secret management extensions.
+
+## Syntax
+
+```Syntax
+secret(<name>)
+secret(<name>, <vault>)
+```
+
+## Description
+
+The `secret()` function retrieves secrets from extensions that support the
+secret capability. It queries all registered extensions that implement secret
+management and returns the requested secret value. If multiple extensions
+return different values for the same secret name, an error is returned unless
+a vault is specified to disambiguate.
+
+The function supports two calling patterns:
+
+- Single argument: Retrieves a secret by name from any available vault
+- Two arguments: Retrieves a secret by name from a specific vault
+
+If multiple extensions return the same secret value, the function succeeds and
+returns that value. This allows for redundancy across secret management
+systems.
+
+## Examples
+
+### Example 1 - Retrieve a secret by name
+
+The following example retrieves a secret named 'DatabasePassword' from any
+available vault. The secret expression is used directly as the output value.
+
+```yaml
+# secret.example.1.dsc.config.yaml
+$schema: https://aka.ms/dsc/schemas/v3/bundled/config/document.json
+resources:
+- name: Database Connection
+  type: Microsoft.DSC.Debug/Echo
+  properties:
+    output: "[secret('DatabasePassword')]"
+```
+
+```bash
+dsc config get --file secret.example.1.dsc.config.yaml
+```
+
+```yaml
+results:
+- name: Database Connection
+  type: Microsoft.DSC.Debug/Echo
+  result:
+    actualState:
+      output: "MySecretPassword123"
```

Comment:
I'm not sure how to address this, but if you _can_ currently use the `Echo` resource to emit secrets retrieved with the `secret()` function, that's probably something we need to fix.

I'm not sure exactly _how_ to manage this problem yet - but if all you need to do to dump secrets from someone's vault is guess the name and invoke a config, that's not good.

CC @SteveL-MSFT

Reply:
If you can access the secret (meaning authenticated in some way), you already have access to the secret in memory (or other means).  Resources need the secret itself in plaintext and we can only prevent accidental exposure of it, we aren't a security boundary to prevent access to something that have to use.

If we want to add a layer of defense, we could say that secrets are always passed as JSON:

```json
{ "secureString": "plaintext" }
```

But now anything that needs it needs to parse the JSON _and_ would also need to handle it appropriately anyways, so I'm not sure it adds value since the recipient would already know it's a sensitive string.